### PR TITLE
chore: update semantic-release, add appveyor

### PR DESCRIPTION
**Pull request by adrukh**
- [x] Ready for review
- [x] Follows CONTRIBUTING rules

#### What does this PR do?

* Update `semantic-release` to catch up with new features

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,18 +8,18 @@
   },
   "main": "lib/index.js",
   "scripts": {
-    "test": "npm run lint && npm run test-functional",
-    "lint": "jscs `find ./lib -name '*.js'` -v && jscs `find ./test -name '*.js'` -v",
-    "test-functional": "tap `find ./test/functional -name '*.test.js'`",
-    "test-system": "tap `find ./test/system -name '*.test.js'`",
-    "semantic-release": "semantic-release pre && npm publish && semantic-release post"
+    "lint": "npm run eslint",
+    "eslint": "./node_modules/.bin/eslint -c .eslintrc lib test",
```
Comment (michael-go) [inline package.json]: doesnt it work just with `eslint`, without the relative path? as done with `tap` below

File: package.json [hunk 2]
```diff
@@ -8,18 +8,18 @@
...
   "author": "snyk.io",
   "license": "Apache-2.0",
   "devDependencies": {
-    "jscs": "^3.0.7",
-    "semantic-release": "^6.3.6",
-    "tap": "^10.3.2",
-    "tap-only": "0.0.5"
+    "eslint": "^5.3.0",
+    "semantic-release": "^15",
+    "tap": "^12.0.1"
```
Comment (michael-go) [inline package.json]: I love `tap-only` - would prefer to leave it. It's useful when debugging - can easily make a single test to run without commenting out code

File: test/functional/mvn-plugin.test.js [hunk 3]
```diff
@@ -1,4 +1,4 @@
-var test = require('tap-only');
+var test = require('tap').test;
```
Comment (michael-go) [inline test/functional/mvn-plugin.test.js]: I love tap-only - would prefer to leave it. It's useful when debugging - can easily make a single test to run without commenting out code

File: package.json (hunk 1 above)
Comment (adrukh) [reply package.json]: yeah totally, will change

File: test/functional/mvn-plugin.test.js (hunk 3 above)
Comment (adrukh) [reply test/functional/mvn-plugin.test.js]: ok let me bring it back, one sec

Comment (michael-go) [reply test/functional/mvn-plugin.test.js]: 🙏

**Pull request merged**